### PR TITLE
add explanation for clearing shared segments

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
As requested on #6530.

### CODE REVIEW

File: ci/clear-shared-segments-macos.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# as long as there is at least one process connected to them. Since PostgreSQL,
+# at least in the configuration we are using, runs from a single process, this
+# is safe to run even while PostgreSQL is still running (Works On My Machine®).
+# Segments do get deleted immediately if there is no process connected to them
+# when the command is run.
+
+steps:
+  - bash: |
+      set -euo pipefail
+      for shmid in $(ipcs -m | sed 1,3d | awk '{print $2}' | sed '$d'); do ipcrm -m $shmid; done
```
Comment (SamirTalwar-DA) [inline ci/clear-shared-segments-macos.yml:31]: I'd find this easier to read if the important operation was on a new line.

```suggestion
      for shmid in $(ipcs -m | sed 1,3d | awk '{print $2}' | sed '$d'); do
        ipcrm -m $shmid
      done
```

**Comment by SamirTalwar-DA** [comment]
Also what @cocreature said. (He's too fast.)

**Comment by SamirTalwar-DA** [comment]
FWIW, I still think this is a good improvement.

**Comment by garyverhaegen-da** [comment]
> LGTM but I understood @SamirTalwar’s suggestion as turning this into a shell script so you can easily run it locally not a yaml file.

Ah, this was actually addressing your comment. I hadn't seen @SamirTalwar-DA's 😓. Will get on that then.

**Pull request merged**